tests/helpers/test_check_config.py

```diff
@@ -271,6 +271,26 @@ async def test_component_platform_import_error(hass: HomeAssistant) -> None:
         _assert_warnings_errors(res, [error], [])
 
 
+async def test_component_platform_import_error(hass: HomeAssistant) -> None:
+    """Test errors if component or platform not found."""
+    # Make sure they don't exist
+    files = {YAML_CONFIG_FILE: BASE_CONFIG + "light:\n  platform: demo"}
+    with patch(
+        "homeassistant.loader.Integration.get_platform",
+        side_effect=[None, ImportError("blablabla")],
+    ), patch("os.path.isfile", return_value=True), patch_yaml_files(files):
+        res = await async_check_ha_config_file(hass)
+        log_ha_config(res)
+
+        assert res.keys() == {"homeassistant", "light"}
+        error = CheckConfigError(
+            "Platform error light.demo - blablabla",
+            None,
+            None,
+        )
+        _assert_warnings_errors(res, [], [error])
```

Comment:
errors is the second parameter and warnings the last parameter. I'd rename the `error` variable to `warning`.